src/lib/proof-system/cache.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  writeFileSync,
+  readFileSync,
+  mkdirSync,
+  resolve,
+  cacheDir,
+} from '../util/fs.js';
+import { jsEnvironment } from '../../bindings/crypto/bindings/env.js';
+
+export { Cache, CacheHeader, cacheHeaderVersion };
+
+/**
+ * Interface for storing and retrieving values, for caching.
+ * `read()` and `write()` can just throw errors on failure.
+ */
+type Cache = {
```

Comment:
this file contains the main new interface -- `Cache`! plus a default implementation of `Cache` which works in Node.js and uses a common cache directory